lynx/common/enitity.py

```diff
@@ -1,6 +1,33 @@
+from __future__ import annotations
+from dataclasses import dataclass
+
 from lynx.common.serializable import Serializable
 
+
 # Right now it just represents common root
 # for both `Object` and `Action`
 class Entity(Serializable):
-    pass
\ No newline at end of file
+    @dataclass
+    class Exported(Serializable):
+        type: str = ""
+        attributes: str = ""
+
+    def get_type(self) -> str:
+        return type(self).__name__
+
+    def get_attributes(self) -> str:
+        return super().serialize()
+
+    def serialize(self) -> str:
+        return self.Exported(self.get_type(), self.get_attributes()).serialize()
+
+    @classmethod
+    def deserialize(cls, json_string: str) -> Entity:
```

Comment:
why `cls` instead of `self`

Reply:
It's a class method. It belongs to the class type, so there is no instance which would be `self`. The `cls` represents the type class such as `Object` which can be used to call its constructor.